app/views/results/common/_file_selector_dropdown.html.erb

```diff
@@ -16,6 +16,18 @@
         <% else %>
         first_file_id = <%= files[0].id %>;
         first_file_path = '<%= files[0].path %>/<%= files[0].filename %>';
+        <% if @current_user.ta? %>
+          <% if assignmentfile =
+              AssignmentFile.select(:filename)
+              .joins('JOIN criteria_assignment_files_joins c ON c.assignment_file_id = assignment_files.id')
+              .joins('JOIN criterion_ta_associations t ON c.criterion_id = t.criterion_id')
+              .where('t.ta_id = ? AND assignment_files.assignment_id = ?', @current_user.id, @assignment.id)
+              .first %>
+            <% index = files.find_index{|x| x.filename == assignmentfile.filename} %>
```

Comment:
This seems like it would break if a student didn't actually submit the required file.